app/models/host/discovered.rb

```diff
@@ -247,4 +247,20 @@ def create_notification
   def delete_notification
     ForemanDiscovery::UINotifications::DestroyHost.deliver!(self)
   end
+
+  def notification_recipients_ids
+    organization_users = if organization.ignore_types.include? 'User'
+                           User.unscoped.all
+                         else
+                           organization.users
+                         end
+
+    org_recipients = organization_users.find_all do |user|
+      user.allowed_to? "create_hosts"
```

Comment:
I think `user.can?` would be a better function, this only checks permissions for a given controller action